src/e2e_test/adaptive_export_loadtest/tools/loadgen/Dockerfile

```
@@ -0,0 +1,18 @@
+# Build both load-test binaries (cleanloadgen + httpsink) into one small image.
+# Built on the PG dev-machine (native amd64 docker), never on the agent VM
+# (ARM, no bazel/heavy-build) — same path used for the dx images.
+FROM golang:1.22-bookworm AS build
+WORKDIR /src
+# -mod=mod lets the build resolve + record go.sum for the single lib/pq dep
+# without a pre-committed go.sum (keeps the tool tree minimal).
+ENV GOFLAGS=-mod=mod CGO_ENABLED=0
+COPY go.mod ./
+COPY cmd ./cmd
+RUN go build -trimpath -ldflags="-s -w" -o /out/cleanloadgen ./cmd/cleanloadgen
+RUN go build -trimpath -ldflags="-s -w" -o /out/httpsink     ./cmd/httpsink
```

Hadolint (2.14.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)